src/utils.js

```diff
@@ -524,46 +541,85 @@ go.utils = {
+            response.sub_rate = im.config.rate.two_per_week;
+        }
+        return response;
+    },
+
+    get_edd: function(im, contact) {
+        // Return estimated due date YYYYMMDD, default to null if unknown
+        // Django should default to 17000101 if null entry given
+        if (!_.isUndefined(contact.extra.due_date_month) &&
+            !_.isUndefined(contact.extra.due_date_day)) {
+            var day = contact.extra.due_date_day;
+            var month = contact.extra.due_date_month;
+            var year = go.utils.get_due_year_from_month(month, go.utils.get_today(im.config));
+            return [year, month, day].join('-');
+        } else {
+            return null;
+        }
     },
 
-    subscription_send_doc: function(contact, im, metric_prefix, env, opts) {
+    get_identification_no: function(contact) {
+        if (contact.extra.id_type === 'sa_id') {
+            return contact.extra.sa_id;
+        } else if (contact.extra.id_type === 'passport') {
+            return contact.extra.passport_no;
+        } else {
+            return null;
+        }
+    },
+
+    get_hcw_no: function(device_msisdn, contact_msisdn) {
+        var n_device_msisdn = go.utils.normalize_msisdn(device_msisdn, '27');
+        var n_contact_msisdn = go.utils.normalize_msisdn(contact_msisdn, '27');
+        if (n_device_msisdn === n_contact_msisdn) {
+            return null;
+        } else {
+            return n_device_msisdn;
+        }
+    },
+
+    post_registration: function(device_msisdn, contact, im, reg_type) {
         var payload = {
-          contact_key: contact.key,
-          lang: contact.extra.language_choice,
-          message_set: "/api/v1/message_set/" + opts.sub_type + "/",
-          next_sequence_number: opts.sub_seq_start,
-          schedule: "/api/v1/periodic_task/" + opts.sub_rate + "/",
-          to_addr: contact.msisdn,
-          user_account: contact.user_account
+            hcw_msisdn: go.utils.get_hcw_no(device_msisdn, contact.msisdn),  // +27...
+            mom_msisdn: go.utils.normalize_msisdn(contact.msisdn, '27'),  // +27...
+            mom_id_type: contact.extra.id_type,  // 'sa_id' | 'passport' | 'none'
+            mom_lang: contact.extra.language_choice,  // 'en' | 'af' | 'xh' ...
+            mom_edd: go.utils.get_edd(im, contact),  // 'YYYY-MM-DD' | null
+            mom_id_no: go.utils.get_identification_no(contact),
+            mom_dob: contact.extra.dob || null,  // 'YYYY-MM-DD' | null
+            clinic_code: contact.extra.clinic_code || null,
+            authority: reg_type,  // 'clinic' | 'chw' | 'personal'
+        };
+        return go.utils
+            .control_api_call("post", null, payload, 'registration/', im)
+            .then(function(post_response) {
+                return go.utils.json_success_fail_metric(im, 'registration_call', post_response);
+            });
+    },
+
+    post_subscription: function(contact, im, metric_prefix, env, opts) {
```

Comment:
@imsickofmaps This looks unclear in the review area, I actually just renamed `subscription_send_doc` to `post_subscription` for clarity.